src/planner/ostree.rs

```diff
@@ -0,0 +1,406 @@
+use crate::{
+    action::{
+        base::{CreateDirectory, CreateFile, RemoveDirectory},
+        common::{ConfigureInitService, ConfigureNix, CreateUsersAndGroups, ProvisionNix},
+        linux::{ProvisionSelinux, StartSystemdUnit, SystemctlDaemonReload},
+        StatefulAction,
+    },
+    error::HasExpectedErrors,
+    planner::{Planner, PlannerError},
+    settings::CommonSettings,
+    settings::{InitSettings, InitSystem, InstallSettingsError},
+    Action, BuiltinPlanner,
+};
+use std::{
+    collections::HashMap,
+    path::{Path, PathBuf},
+};
+use tokio::process::Command;
+use which::which;
+
+use super::ShellProfileLocations;
+
+/// A planner for Linux installs
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+#[cfg_attr(feature = "cli", derive(clap::Parser))]
+pub struct Ostree {
+    /// Where `/nix` will be bind mounted to.
+    #[cfg_attr(feature = "cli", clap(long, default_value = "/var/home/nix"))]
+    persistence: PathBuf,
+    #[cfg_attr(feature = "cli", clap(flatten))]
+    pub settings: CommonSettings,
+    #[cfg_attr(feature = "cli", clap(flatten))]
+    pub init: InitSettings,
```

Comment:
I don't think this `ostree` planner can support `--init none` (similar to how the Steam Deck cannot). It's ok to remove this.